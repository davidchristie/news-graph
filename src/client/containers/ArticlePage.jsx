import React, { PropTypes } from 'react'
import { connect } from 'react-redux'

// import ArticleNetwork from './ArticleNetwork'
import Jumbotron from '../components/Jumbotron'

export const ArticlePage = class ArticlePage extends React.Component {
  render () {
    const url = this.props.location.query.url
    const articles = this.props.articles
    const title = 'Title'
    const description = 'Description'
    // const article = articles[url]
    // const title = article.title
    // const description = article.description
    // <ArticleNetwork article={article} />
    return (
      <div>
        <Jumbotron title={title} lead={description} />
        <div className='card'>
          <div className='card-block'>
            <h2 className='card-title'>Connections</h2>
          </div>
        </div>
      </div>
    )
  }
}

ArticlePage.propTypes = {
  articles: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
  }).isRequired).isRequired,
  location: PropTypes.shape({
    query: PropTypes.shape({
      url: PropTypes.string.isRequired
    }).isRequired
  }).isRequired
}

function mapStateToProps (state) {
  return {
    articles: state.app.articles
  }
}

export default connect(mapStateToProps)(ArticlePage)
