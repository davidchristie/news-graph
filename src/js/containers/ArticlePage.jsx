import React, { PropTypes } from 'react'
import { connect } from 'react-redux'

import ArticleNetwork from './ArticleNetwork'

const ArticlePage = class ArticlePage extends React.Component {
  render() {
    const id = Number(this.props.params.id)
    const articles = this.props.articles
    const article = articles.find(article => article.id === id)
    const title = article.title
    return (
      <div>
        <h1>{title}</h1>
        <div className="row">
          <div className="card col-md-12">
            <div className="card-block">
              <h2 className="card-title">Connections</h2>
              <ArticleNetwork article={article} />
            </div>
          </div>
        </div>
      </div>
    )
  }
}

ArticlePage.propTypes = {
  articles: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
  }).isRequired).isRequired,
  params: PropTypes.shape({
    id: PropTypes.string.isRequired
  }).isRequired
}

function mapStateToProps(state) {
  return {
    articles: state.app.articles
  }
}

export default connect(mapStateToProps)(ArticlePage)
