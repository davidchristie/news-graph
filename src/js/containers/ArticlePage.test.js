/* eslint-env jest */

import React from 'react'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'

import ArticlePage from './ArticlePage'
import initialState from '../examples/initial-state'

const store = configureStore()(initialState)
const id = 1
const article = initialState.app.articles.find(article => article.id === id)

test('ArticlePage renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <ArticlePage article={article} params={{id: String(id)}} />
    </Provider>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
