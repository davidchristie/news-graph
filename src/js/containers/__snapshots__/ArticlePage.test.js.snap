exports[`test ArticlePage renders correctly 1`] = `
<div>
  <h1>
    Ancient retroviruses emerged half a billion years ago : Nature News & Comment
  </h1>
  <div
    className="row">
    <div
      className="card col-md-12">
      <div
        className="card-block">
        <h2
          className="card-title">
          Connections
        </h2>
        <div
          style={
            Object {
              "backgroundColor": "#333333",
              "border": "1px solid lightgray",
              "height": "400px",
              "width": "100%",
            }
          } />
      </div>
    </div>
  </div>
</div>
`;
